<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>870</width>
    <height>638</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="mouseTracking">
   <bool>false</bool>
  </property>
  <property name="windowTitle">
   <string>Stochastic L-system</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/new/prefix1/icon.ico</normaloff>:/new/prefix1/icon.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="styleSheet">
    <string notr="true">QWidget#centralWidget{
background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 rgba(240,249,255), stop:0.93 rgba(228,228,228),stop:1.0 rgba(237,250,250,1));
}</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="Mainbase" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true">QWidget{
margin:0px;
padding:0px;
background:none;
}</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QWidget" name="widget" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">QWidget{
background:none;
}</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="DisplayWidget" name="openGLWidget">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>571</width>
              <height>531</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="ControlPanel" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QTabWidget" name="tabWidget">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>221</width>
                 <height>511</height>
                </size>
               </property>
               <property name="styleSheet">
                <string notr="true">QTabWidget::pane{ /* The tab widget frame */
background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 rgba(240,249,255), stop:0.93 rgba(228,228,228),stop:1.0 rgba(237,250,250,1));
}
QTabWidget::pane {
    border-top: 1px solid #C2C7CB;
	border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
}

QTabBar::tab {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #f6f8f9 , stop: 0.23 #e5ebee,
                                stop: 0.25 #d7dee3, stop: 1.0 #f5f7f9);
    border: 1px solid #C4C4C3;
 /* same as the pane color */
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
border-bottom-width:0px;
    min-width: 8ex;
    padding: 4px;
}
QTabBar::tab:selected,::tab::hover{
	background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #e0f3fa, stop: 0.23 #d8f0fc ,
                                stop: 0.25 #b8e2f6, stop: 1.0 #b6dffd);
}</string>
               </property>
               <property name="currentIndex">
                <number>0</number>
               </property>
               <widget class="QWidget" name="tab_9">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <attribute name="title">
                 <string>L-system</string>
                </attribute>
                <widget class="QLabel" name="label">
                 <property name="geometry">
                  <rect>
                   <x>30</x>
                   <y>10</y>
                   <width>31</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>File :</string>
                 </property>
                </widget>
                <widget class="QComboBox" name="FileSelector">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>10</y>
                   <width>111</width>
                   <height>22</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Select examples&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="statusTip">
                  <string>change file template</string>
                 </property>
                 <property name="whatsThis">
                  <string>Choose file template</string>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QComboBox{
border-radius:5px;
border: 1px solid grey;
}
QComboBox:!editable:on, QComboBox::drop-down:editable:on {
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #D3D3D3, stop: 0.4 #D8D8D8,
                                stop: 0.5 #DDDDDD, stop: 1.0 #E1E1E1);
}

QComboBox::drop-down {
border-top-right-radius:5px;
border-bottom-right-radius:5px;
background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                stop: 0 #ebf1f6, stop: 0.5 #abd3ee ,
                                stop: 0.51 #89c3eb, stop: 1.0 #d5ebfb);
}</string>
                 </property>
                 <item>
                  <property name="text">
                   <string>(none)</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Example1</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Example2</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>Example3</string>
                  </property>
                 </item>
                </widget>
                <widget class="QLabel" name="label_2">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>40</y>
                   <width>41</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Axiom:</string>
                 </property>
                </widget>
                <widget class="QPushButton" name="foceUpdateButton">
                 <property name="geometry">
                  <rect>
                   <x>120</x>
                   <y>440</y>
                   <width>81</width>
                   <height>31</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Update display&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPushButton{

border:1px solid grey;
border-radius: 5px;
color:white;
background-color: qlineargradient(spread:pad,x1:0 y1:0, x2:0 y2:1, stop:0 rgba(174,188,191), stop:0.50 rgba(110,119,116),stop:0.51 rgba(20, 20, 20, 255),stop:1 rgba(60, 60, 60, 255));
}

QPushButton:hover{
background-color: qlineargradient(spread:pad,x1:0 y1:0, x2:0 y2:1, stop:0 rgba(53,148,198) , stop:0.50 rgba(18,63,86),stop:0.51 rgba(10,14,10),stop:1 rgba(6,103,130, 255));
}

QPushButton:
</string>
                 </property>
                 <property name="text">
                  <string>Update</string>
                 </property>
                </widget>
                <widget class="QLineEdit" name="Axiom">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>40</y>
                   <width>113</width>
                   <height>20</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Change Axiom&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLineEdit{
border:1px solid rgba(200,200,200);
border-radius:4px;
}
QLineEdit::focus{
border:1px solid blue;
}</string>
                 </property>
                </widget>
                <widget class="QLabel" name="label_3">
                 <property name="geometry">
                  <rect>
                   <x>10</x>
                   <y>70</y>
                   <width>61</width>
                   <height>21</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Iteration:</string>
                 </property>
                </widget>
                <widget class="QSpinBox" name="Iteration_spinBox">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>70</y>
                   <width>111</width>
                   <height>22</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Change Iteration Level&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QSpinBox{
background-color:qlineargradient(spread:pad,x1:0 y1:0, x2:0 y2:1, stop:0 rgba(80, 80, 80), stop:0.16 rgba(160, 160, 160),stop:0.24 rgba(220, 220, 220, 255),stop:1 rgba(210, 210, 210, 255));
border-radius:3px;
}
QSpinBox::up-button {

     width: 20px; /* 16 + 2*1px border-width = 15px padding + 3px parent border */
	border-top-right-radius:3px;
background-color: qlineargradient(spread:pad,x1:0 y1:0, x2:0 y2:1, stop:0 gba(30,87,153,1),stop:0.5 rgba(41,137,216,1),stop:0.51 rgba(32,124,202,1) ,stop:1.0 rgba(125,185,232,1));
 }

 QSpinBox::up-button:hover {
 }

 QSpinBox::up-button:pressed {
 }

 QSpinBox::up-arrow {
     width: 7px;
     height: 7px;
 }

 QSpinBox::down-button {
    border-bottom-right-radius:3px;
	width: 20px;
	background-color: qlineargradient(spread:pad,x1:0 y1:0, x2:0 y2:1, stop:0 gba(30,87,153,1),stop:0.5 rgba(41,137,216,1),stop:0.51 rgba(32,124,202,1) ,stop:1.0 rgba(125,185,232,1));
 }

 QSpinBox::down-button:hover {
 }

 QSpinBox::down-button:pressed {
	
 }

 QSpinBox::down-arrow {
     width: 7px;
     height: 7px;
 }</string>
                 </property>
                </widget>
                <widget class="QPlainTextEdit" name="RuleDIsplay">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>130</y>
                   <width>141</width>
                   <height>131</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Shows rules&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPlainTextEdit{
border-radius:4px;
background-color:#fafafa;
border-width:1px;
}</string>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                </widget>
                <widget class="QLabel" name="label_4">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>130</y>
                   <width>31</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Rules:</string>
                 </property>
                </widget>
                <widget class="QLabel" name="label_7">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>100</y>
                   <width>41</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Angle:</string>
                 </property>
                </widget>
                <widget class="QLineEdit" name="AngleInput">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>100</y>
                   <width>113</width>
                   <height>20</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLineEdit{
border:1px solid rgba(200,200,200);
border-radius:4px;
}
QLineEdit::focus{
border:1px solid blue;
}</string>
                 </property>
                </widget>
                <widget class="QLabel" name="label_8">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>280</y>
                   <width>41</width>
                   <height>20</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Result:</string>
                 </property>
                </widget>
                <widget class="QPlainTextEdit" name="L_systemResult">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>280</y>
                   <width>141</width>
                   <height>121</height>
                  </rect>
                 </property>
                 <property name="toolTip">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show results&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPlainTextEdit{
border-radius:3px;
}</string>
                 </property>
                 <property name="readOnly">
                  <bool>true</bool>
                 </property>
                 <property name="overwriteMode">
                  <bool>false</bool>
                 </property>
                </widget>
               </widget>
               <widget class="QWidget" name="tab_2">
                <attribute name="title">
                 <string>Rendering</string>
                </attribute>
                <widget class="QLabel" name="label_5">
                 <property name="geometry">
                  <rect>
                   <x>10</x>
                   <y>90</y>
                   <width>47</width>
                   <height>13</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Texture:</string>
                 </property>
                </widget>
                <widget class="QGraphicsView" name="graphicsView">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>90</y>
                   <width>151</width>
                   <height>71</height>
                  </rect>
                 </property>
                </widget>
                <widget class="QLabel" name="label_6">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>170</y>
                   <width>47</width>
                   <height>13</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QLabel{
background:none;
}</string>
                 </property>
                 <property name="text">
                  <string>Model:</string>
                 </property>
                </widget>
                <widget class="QTableView" name="tableView">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>220</y>
                   <width>151</width>
                   <height>192</height>
                  </rect>
                 </property>
                 <property name="frameShadow">
                  <enum>QFrame::Sunken</enum>
                 </property>
                 <property name="lineWidth">
                  <number>0</number>
                 </property>
                 <property name="sortingEnabled">
                  <bool>false</bool>
                 </property>
                </widget>
                <widget class="QRadioButton" name="radioButton">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>170</y>
                   <width>82</width>
                   <height>17</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Use Default</string>
                 </property>
                </widget>
                <widget class="QRadioButton" name="radioButton_2">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>190</y>
                   <width>82</width>
                   <height>17</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Use modelList</string>
                 </property>
                </widget>
                <widget class="QLabel" name="label_9">
                 <property name="geometry">
                  <rect>
                   <x>20</x>
                   <y>430</y>
                   <width>181</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Warning: this tab is useless for now</string>
                 </property>
                </widget>
                <widget class="QLabel" name="label_10">
                 <property name="geometry">
                  <rect>
                   <x>10</x>
                   <y>10</y>
                   <width>47</width>
                   <height>13</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Color:</string>
                 </property>
                </widget>
                <widget class="QWidget" name="color_display" native="true">
                 <property name="geometry">
                  <rect>
                   <x>50</x>
                   <y>10</y>
                   <width>81</width>
                   <height>21</height>
                  </rect>
                 </property>
                 <property name="styleSheet">
                  <string notr="true"/>
                 </property>
                </widget>
                <widget class="QPushButton" name="ColorSelector">
                 <property name="geometry">
                  <rect>
                   <x>130</x>
                   <y>10</y>
                   <width>51</width>
                   <height>23</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>select</string>
                 </property>
                </widget>
               </widget>
               <widget class="QWidget" name="tab">
                <attribute name="title">
                 <string>Rules</string>
                </attribute>
                <widget class="QLabel" name="label_11">
                 <property name="geometry">
                  <rect>
                   <x>10</x>
                   <y>60</y>
                   <width>47</width>
                   <height>13</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>AddRule:</string>
                 </property>
                </widget>
                <widget class="QLabel" name="label_12">
                 <property name="geometry">
                  <rect>
                   <x>10</x>
                   <y>130</y>
                   <width>61</width>
                   <height>16</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>CreateRule:</string>
                 </property>
                </widget>
                <widget class="QLineEdit" name="lineEdit">
                 <property name="geometry">
                  <rect>
                   <x>60</x>
                   <y>60</y>
                   <width>113</width>
                   <height>20</height>
                  </rect>
                 </property>
                </widget>
                <widget class="QPushButton" name="BrowseRuleFile">
                 <property name="geometry">
                  <rect>
                   <x>170</x>
                   <y>60</y>
                   <width>41</width>
                   <height>23</height>
                  </rect>
                 </property>
                 <property name="text">
                  <string>Open</string>
                 </property>
                </widget>
                <widget class="QPlainTextEdit" name="plainTextEdit">
                 <property name="geometry">
                  <rect>
                   <x>30</x>
                   <y>150</y>
                   <width>171</width>
                   <height>211</height>
                  </rect>
                 </property>
                </widget>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>870</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionReset"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionDocument"/>
    <addaction name="actionL_system"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="statusTip">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionDocument">
   <property name="text">
    <string>document</string>
   </property>
   <property name="statusTip">
    <string>document</string>
   </property>
  </action>
  <action name="actionL_system">
   <property name="text">
    <string>L-system</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>DisplayWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">displaywidget.h</header>
   <slots>
    <slot>changeAngle()</slot>
    <slot>Update()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
